### PR TITLE
Use name in the code when generating new linera project

### PR DESCRIPTION
**Pull request by deuszx**
## Motivation

When generating a new linera project we were using generic `Application` as the name for `struct`s, `trait`s, etc rather than given project name (used when calling `linera project new <name>` command).

## Proposal

Parse given project name to PascalCase and use where appropriate.

Closes https://github.com/linera-io/linera-protocol/issues/2577
## Test Plan

Built the `linera` binary locally and run `linera project new my-counter`. The project builds correctly (after fixing dependencies' versions) and and the `my-counter` is used properly:
* `my_counter` as a module name.
* `MyCounter` as a name in `trait`s and `struct`s.

## Release Plan

- Nothing to do / These changes follow the usual release cycle.
## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-service/src/project.rs [hunk 1]
```diff
@@ -230,6 +242,26 @@ impl Project {
...
+                result.push(c.to_ascii_uppercase());
+                // If we capitalize a letter, we should not capitalize the next one.
+                capitalize = false;
+            } else {
+                result.push(c);
+            }
+        }
+        result
+    }
+
```
Comment (Twey) [inline linera-service/src/project.rs:264]: We already have the [https://docs.rs/convert_case/latest/convert_case/](convert_case) crate as a dependency — maybe we should use that?

File: linera-service/src/project.rs [hunk 2]
```diff
@@ -230,6 +242,26 @@ impl Project {
         (linera_sdk_dep, linera_sdk_dev_dep)
     }
```
Comment (Twey) [inline linera-service/src/project.rs:243]: I think that we could replace these functions with a call to [`cargo-generate`](https://github.com/cargo-generate/cargo-generate).  It [supports being called as a library](https://github.com/cargo-generate/cargo-generate/blob/main/examples/how-to-use-cargo-gen-as-library/src/main.rs) and handles all the substitution for you — complete with a nice developer flow where it prompts for information that isn't provided.

File: linera-service/src/project.rs (hunk 1 above)
Comment (deuszx) [reply linera-service/src/project.rs]: Done in [6e150d1](https://github.com/linera-io/linera-protocol/pull/2578/commits/6e150d1648d4ce6f020e7cf77c3d65baf1f43b4e)

File: linera-service/src/project.rs (hunk 2 above)
Comment (deuszx) [reply linera-service/src/project.rs]: Lifted to an issue https://github.com/linera-io/linera-protocol/issues/2579.

**Review by christos-h — APPROVED** [review]
Awesome!

**Pull request merged**